petab/measurements.py

```diff
@@ -212,8 +213,19 @@ def split_parameter_replacement_list(
             return []
         return [list_string]
 
-    result = [x.strip() for x in list_string.split(delim) if len(x.strip())]
-    return [core.to_float_if_float(x) for x in result]
+    result = [x.strip() for x in list_string.split(delim)]
+
+    def convert_and_check(x):
+        x = core.to_float_if_float(x)
+        if isinstance(x, float):
+            return x
+        if lint.is_valid_identifier(x):
+            return x
+        raise ValueError(f"'{x}' in parameter replacement list "
+                         f"'{list_string}' is neither a number, nor a "
+                         "valid parameter ID.")
```

Comment:
`x` has had `strip()` applied to it, so may be confusing to see possibly just `''`.

```suggestion
        raise ValueError(
            f"The value '{x}' (whitespace removed) in the parameter "
            f"replacement list '{list_string}' is neither a number, nor a "
            "valid parameter ID."
        )
```

Reply:
Ah, will remove the `strip()`. We don't want that here.